x-pack/plugins/upgrade_assistant/server/lib/es_system_indices_upgrade.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { flow, flatMap, map, flatten, uniq } from 'lodash/fp';
+import { ElasticsearchClient } from 'src/core/server';
+import { SystemIndicesUpgradeStatus, SystemIndicesUpgradeFeature } from '../../common/types';
+
+export const convertFeaturesToIndicesArray = (
+  features: SystemIndicesUpgradeFeature[]
+): string[] => {
+  return flow(
+    // Map each feature into Indices[]
+    map('indices'),
+    // Flatten each into an string[] of indices
+    map(flatMap('index')),
+    // Flatten the array
+    flatten,
+    // And finally dedupe the indices
+    uniq
+  )(features);
+};
+
+export const getESSystemIndicesUpgradeStatus = async (
+  client: ElasticsearchClient
+): Promise<SystemIndicesUpgradeStatus> => {
+  const { body } = await client.transport.request({
```

Comment:
I've created an issue https://github.com/elastic/elasticsearch-js/issues/1561 for the elasticsearch-js team to add the missing function

Reply:
In the docs page there's an unresolved todo item to verify something about [a possibly required permission](https://www.elastic.co/guide/en/elasticsearch//reference/master/migration-api-feature-upgrade.html#feature-upgrade-api-prereqs). I've asked the ES team for clarification on that and if required we can add a check for it in a subsequent PR.